### PR TITLE
Expand pre commit and ci

### PR DESCRIPTION
**Pull request by paddyroddy**
## Description
**What is this pull request (PR)**
- [ ] Bug fix
- [X] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
To ensure the quality of the code remains high as well as ease the ability to review PRs as all formatting problems are taken care of

**What does this PR do?**
It expands the existing set of `pre-commit` hooks as well as isolating the linting to a separate CI

## References
A copy of https://github.com/brainglobe/cellfinder-napari/pull/58 and https://github.com/brainglobe/cellfinder-core/pull/29

## How has this PR been tested?
In a later PR (to be raised) the `pre-commit` hooks are run on all bits of existing code

## Is this a breaking change?
Technically yes. This specific PR will introduce a new CI which will fail, but a subsequent PR will fix the issue.

## Does this PR require an update to the documentation?
No other than potentially a warning to developers that they must pass `pre-commit` otherwise a CI element will fail

## Checklist:
- [X] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [ ] The documentation has been updated to reflect any changes
- [X] The code has been formatted with [Black](https://github.com/psf/black)

### CODE REVIEW

File: .github/workflows/test_and_deploy.yml [hunk 1]
```diff
@@ -7,24 +7,7 @@ env:
   IMAGE_NAME: ${{ github.repository }}
   
 jobs:
```
Comment (adamltyson) [inline .github/workflows/test_and_deploy.yml:9]: See question on `cellfinder-core`, can we keep the lint -> test -> deploy dependency?

File: README.md [hunk 2]
```diff
@@ -10,6 +10,9 @@
 [![Contributions](https://img.shields.io/badge/Contributions-Welcome-brightgreen.svg)](https://docs.brainglobe.info/cellfinder/contributing)
 [![Website](https://img.shields.io/website?up_message=online&url=https%3A%2F%2Fbrainglobe.info)](https://brainglobe.info/cellfinder)
 [![Twitter](https://img.shields.io/twitter/follow/brain_globe?style=social)](https://twitter.com/brain_globe)
+[![Imports: isort](https://img.shields.io/badge/%20imports-isort-%231674b1?style=flat&labelColor=ef8336)](https://pycqa.github.io/isort/)
```
Comment (adamltyson) [inline README.md:13]: See comment on `cellfinder-core`. Can we move these new badges to before `Contributions`?

File: .github/workflows/test_and_deploy.yml (hunk 1 above)
Comment (paddyroddy) [reply .github/workflows/test_and_deploy.yml:9]: yep will look into this on both

File: README.md (hunk 2 above)
Comment (paddyroddy) [reply README.md:13]: yep will do

**Pull request merged**